server/src/main/java/com/github/animelist/animelist/model/profilepage/UserListBlock.java

```diff
@@ -17,6 +18,13 @@ public UserListBlock(Width width, BlockType type,
         Assert.isTrue(type == BlockType.USER_LIST, "block type must match class");
     }
 
+    @PersistenceConstructor
```

Comment:
What does this persistence constructor annotation do?

Reply:
for some reason (likely because these classes have multiple constructors, and/or because these classes are nested) MongoDB was throwing this error when attempting to rehydrate the User type from the stored document: https://stackoverflow.com/questions/55827640/how-to-fix-failed-to-instantiate-classname-using-constructor-no-constructor

so this annotation tells mongo to use that constructor when rehydrating